src/pudl/settings.py

```diff
@@ -243,41 +244,32 @@ class Eia860Settings(GenericDatasetSettings):
         data_source: DataSource metadata object
         years: list of years to validate.
         eia860m: whether or not to incorporate an EIA-860m month.
-        eia860m_year_month ClassVar[str]: The 860m year-month to incorporate.
+        eia860m_year_months ClassVar[str]: The 860m year-month to incorporate.
     """
 
     data_source: ClassVar[DataSource] = DataSource.from_id("eia860")
     years: list[int] = data_source.working_partitions["years"]
-
     eia860m: bool = True
-    eia860m_year_month: ClassVar[str] = max(
-        DataSource.from_id("eia860m").working_partitions["year_months"]
-    )
+    eia860m_year_months: list[str] = []
 
-    @field_validator("eia860m")
+    @field_validator("eia860m_year_months")
     @classmethod
-    def check_eia860m_year_month(cls, eia860m: bool) -> bool:
-        """Check 860m date-year is exactly one year after most recent working 860 year.
-
-        Args:
-            eia860m: True if 860m is requested.
-
-        Returns:
-            eia860m: True if 860m is requested.
-
-        Raises:
-            ValueError: the 860m date is within 860 working years.
-        """
-        eia860m_year = pd.to_datetime(cls.eia860m_year_month).year
-        expected_year = max(cls.data_source.working_partitions["years"]) + 1
-        if eia860m and (eia860m_year != expected_year):
-            raise AssertionError(
-                f"Attempting to integrate an eia860m year "
-                f"({eia860m_year}) from {cls.eia860m_year_month} not immediately following"
-                f"the eia860 years: {cls.data_source.working_partitions['years']}. "
-                f"Consider switching eia860m parameter to False."
-            )
-        return eia860m
+    def add_other_860m_years(cls, v, info: FieldValidationInfo) -> list[str]:
+        """Find and assign years from EIA860m if applicable."""
+        if info.data["eia860m"]:
+            all_eia860m_dates = DataSource.from_id("eia860m").working_partitions[
+                "year_months"
+            ]
+            years_eia860m = {date.split("-")[0] for date in all_eia860m_dates}
+            extra_years_eia860m = [
+                year for year in years_eia860m if int(year) not in info.data["years"]
+            ]
+            eia860m_year_months = [
+                max(date for date in all_eia860m_dates if date.startswith(year))
+                for year in extra_years_eia860m
+            ]
+            return eia860m_year_months
+        return v
```

Comment:
I think the way this is setup no matter what you feed into the settings class the same list of year_months will be spit out. which is not ideal.

imo the right conceptual process for this would be very similar to what you are doing but instead of always assigning the same `eia860m_year_months`, you'd check what years are in `v` (which is whatever you assign `eia860m_year_months` when instantiating the class). If there are any gap years between the last `info.data["years"]` and the years in `v`, then add the max 860m date.